sub-packages/bionemo-llm/src/bionemo/llm/lightning.py

```diff
@@ -211,6 +212,34 @@ def predict_loss_reduction(self) -> PassthroughLossReduction:
 """
 
 
+class MegatronPerplexityMetric(torchmetrics.text.Perplexity):
+    def __init__(self, *args, **kwargs):
+        if parallel_state.get_context_parallel_world_size() > 1:
+            raise NotImplementedError(f"{self.__class__} does not support context parallelism yet.")
+
+        self.cross_entropy_loss_fusion = kwargs.pop("cross_entropy_loss_fusion", False)
```

Comment:
My only concern here is that this is a perplexity-specific implementation. Is there no way we can do this without a call to `unreduced_token_loss_fn`?

Reply:
`unreduced_token_loss_fn` is the only tp-aware nll functions we have. Replacing would just be reinventing the wheel.

Reply:
My major concern is a lack of robust testing under parallelism in unittest. We don't have that infrastructure yet.